fiftyone/utils/yolo.py

```diff
@@ -218,24 +514,218 @@ def label_cls(self):
         return fol.Detections
 
     def setup(self):
-        self._obj_names_path = os.path.join(self.export_dir, "obj.names")
-        self._images_path = os.path.join(self.export_dir, "images.txt")
-        self._data_dir = os.path.join(self.export_dir, "data")
+        if self.export_dir is None:
+            rel_dir = self.data_path
+        else:
+            rel_dir = self.export_dir
+
+        self._rel_dir = rel_dir
 
         self._classes = {}
         self._labels_map_rev = {}
         self._images = []
+        self._writer = YOLOAnnotationWriter()
+
+        self._parse_classes()
 
-        self._filename_maker = fou.UniqueFilenameMaker(
-            output_dir=self._data_dir,
+        self._media_exporter = foud.ImageExporter(
+            self.export_media,
+            export_path=self.data_path,
+            supported_modes=(True, False, "move", "symlink"),
```

Comment:
What is the reason that YOLO exporters don't provide a `manifest` option?

Reply:
YOLOv4 stores the labels in the same folder as the images, and YOLOv5 does a find-replace on `/images/` -> `/labels/` to locate the labels coresponding to images, so there's really not a valid use case I could find for trying to support the "manifest" option.

There are a fair number of annoying nuances/peculiarities to consider for each dataset format...

Reply:
In my mind, the "manifest" option is always useful so that you can easily export only labels and then reimport them again directly from that folder. "symlink" provides the same workflow but there may be situations where you would rather have a single manifest file than a directory of symlinks. That being said, it's definitely not quite as useful for YOLO as for other formats so if you want to leave it the way it is now, that's alright by me.

Reply:
Okay I see your point. Let's take this one offline